dts/bindings/sensor/ti,ina237.yaml

```diff
@@ -15,13 +15,80 @@ compatible: "ti,ina237"
 include: ti,ina23x-common.yaml
 
 properties:
+  config:
+    type: int
+    deprecated: true
+    default: 0x0000
+    description: |
+      Value of the configuration register
+      e.g shunt voltage and bus voltage ADC conversion
+      times and averaging, operating mode for INA230 or
+      delay for initial ADC conversion, shunt full scale range
+      for INA237.
+
+      Deprecated, please use new properties instead.
```

Comment:
That brings up the other point that the INA237 and INA230 are really different chips that do not share common configuration values or features other than the original.  For example, the INA230 does not have a temperature sensor, the adc-mode settings are different, and the conversion times are 140, 204, 332, 588, 1100, 2116, 4156, and 8244 us.

Taking a look at the common code between the two (ina23x_common.h,  ina23x_common.c), it really is just the I2C register read/write functions that are common:
```c
int ina23x_reg_read_24(const struct i2c_dt_spec *bus, uint8_t reg, uint32_t *val);
int ina23x_reg_read_16(const struct i2c_dt_spec *bus, uint8_t reg, uint16_t *val);
int ina23x_reg_write(const struct i2c_dt_spec *bus, uint8_t reg, uint16_t val);
```

It seems like the best option here is to get rid of the ina23x* code and create completely independent ina230 and ina237 drivers.  Maybe that is something we could open up as a "Good first issue" task instead of throwing it into this PR?

Reply:
I added the DT configurations to the INA230 to address your concern.  I also added a unit test to make sure I didn't break anything.  Changes will be in the next push to the PR.

Reply:
@MaureenHelm - please take another look when you have a chance.